tests/test_statemachine.py

```diff
@@ -298,3 +298,41 @@ class NoTransitionsMachine(StateMachine):
 
     with pytest.raises(exceptions.InvalidDefinition):
         NoTransitionsMachine()
+
+
+def test_perfectly_fine_machine_should_be_connected(traffic_light_machine):
+    model = MyModel()
+    machine = traffic_light_machine(model)
+    initial_state = [s for s in traffic_light_machine.states if s.initial][0]
+    disconnected_states = machine._disconnected_states(initial_state)
+    assert len(disconnected_states) == 0
+
+
+def test_should_not_create_disconnected_machine():
+    class BrokenTrafficLightMachine(StateMachine):
+        "A broken traffic light machine"
+        green = State('Green', initial=True)
+        yellow = State('Yellow')
+        blue = State('Blue')  # This state is unreachable
+
+        cycle = green.to(yellow) | yellow.to(green)
+
+    with pytest.raises(exceptions.InvalidDefinition):
+        BrokenTrafficLightMachine()
```

Comment:
Since there's a format involved, please also assert the expected message, as in 
https://github.com/fgmacedo/python-statemachine/blob/e3e7409503ed625918b955b92a8aa105d9b6f5ed/tests/test_multiple_destinations.py#L180